Src/WindowsServer/WindowsServer.Shared.Tests/ConfigurationTelemetryInitializerTest.cs

```diff
@@ -0,0 +1,196 @@
+            {
+                RoleName = TestRoleName,
+                RoleInstance = TestRoleInstance,
+            };
+
+            initializer.Initialize(telemetryItem);
+
+            Assert.Equal(TestValue, telemetryItem.Context.Cloud.RoleName);
+            Assert.Equal(TestRoleInstance, telemetryItem.Context.Cloud.RoleInstance);
+            Assert.Equal(TestRoleInstance, telemetryItem.Context.GetInternalContext().NodeName);
+        }
+
+        [TestMethod]
+        public void ConfigurationTelemetryInitializerDoesNotOverrideRoleInstance()
+        {
+            const string TestValue = "Test";
+
+            var telemetryItem = new EventTelemetry();
+            telemetryItem.Context.Cloud.RoleInstance = TestValue;
+
+            var initializer = new ConfigurationTelemetryInitializer
+            {
+                RoleName = TestRoleName,
+                RoleInstance = TestRoleInstance,
+            };
+
+            initializer.Initialize(telemetryItem);
+
+            Assert.Equal(TestRoleName, telemetryItem.Context.Cloud.RoleName);
+            Assert.Equal(TestValue, telemetryItem.Context.Cloud.RoleInstance);
+            Assert.Equal(TestRoleInstance, telemetryItem.Context.GetInternalContext().NodeName);
+        }
+
+        [TestMethod]
+        public void ConfigurationTelemetryInitializerDoesNotOverrideNodeName()
+        {
+            const string TestValue = "Test";
+
+            var telemetryItem = new EventTelemetry();
+            telemetryItem.Context.GetInternalContext().NodeName = TestValue;
+
+            var initializer = new ConfigurationTelemetryInitializer
+            {
+                RoleName = TestRoleName,
+                RoleInstance = TestRoleInstance,
+            };
+
+            initializer.Initialize(telemetryItem);
+
+            Assert.Equal(TestRoleName, telemetryItem.Context.Cloud.RoleName);
+            Assert.Equal(TestRoleInstance, telemetryItem.Context.Cloud.RoleInstance);
+            Assert.Equal(TestValue, telemetryItem.Context.GetInternalContext().NodeName);
+        }
+
+        [TestMethod]
+        public void ConfigurationTelemetryInitializerEmptyVariable()
+        {
+            string roleNameTestVarName, roleInstanceTestVarName;
+            GetAndInitializeEnvironment(out roleNameTestVarName, out roleInstanceTestVarName);
+            ClearEnvironmentVariables(roleNameTestVarName, roleInstanceTestVarName);
+
+            var telemetryItem = new EventTelemetry();
+
+            var initializer = new ConfigurationTelemetryInitializer();
+
+            initializer.Initialize(telemetryItem);
+
+            Assert.Null(telemetryItem.Context.Cloud.RoleName);
+            Assert.Null(telemetryItem.Context.Cloud.RoleInstance);
+            Assert.Null(telemetryItem.Context.GetInternalContext().NodeName);
+        }
+
+        private static void GetAndInitializeEnvironment(out string roleNameTestVarName, out string roleInstanceTestVarName)
```

Comment:
Consider returning an `IDisposable` that does the clear. Then callers can use `using` to ensure clean-up even when exceptions are thrown.

Reply:
Don't you think that's overkill? But using a `try`/`finally` like other tests would be a good idea.